src/frontends/paddle/src/op/pool3d.cpp

```diff
@@ -0,0 +1,202 @@
+        auto_pad = ov::op::PadType::EXPLICIT;
+    }
+
+    /*If pool padding size is a tuple or list, it could be in three forms:
+    [pad_depth, pad_height, pad_width] or [pad_depth_front, pad_depth_back,
+    pad_height_top, pad_height_bottom, pad_width_left, pad_width_right],
+    and when data_format is “NCDHW”, pool_padding can
+    be in the form [[0,0], [0,0], [pad_depth_front, pad_depth_back], [pad_height_top,
+    pad_height_bottom], [pad_width_left, pad_width_right]]. when
+    data_format is “NDHWC”, pool_padding can be in the form
+    [[0,0], [pad_depth_front, pad_depth_back], [pad_height_top,
+    pad_height_bottom], [pad_width_left, pad_width_right], [0,0]].
+    Otherwise, the pool padding size will be a square of an int.*/
+    auto paddings = node.get_attribute<std::vector<int32_t>>("paddings");
+
+    // Default is empty for 'adaptive max pooling'
+    auto data_format = node.get_attribute<std::string>("data_format", {});
+
+    switch (paddings.size()) {
+    case 3:
+        pad_begin =
+            Shape{static_cast<size_t>(paddings[0]), static_cast<size_t>(paddings[1]), static_cast<size_t>(paddings[2])};
+        pad_end = pad_begin;
+        break;
+    case 6:
+        pad_begin =
+            Shape{static_cast<size_t>(paddings[0]), static_cast<size_t>(paddings[2]), static_cast<size_t>(paddings[4])};
+        pad_end = Shape{
+            static_cast<size_t>(paddings[1]),
+            static_cast<size_t>(paddings[3]),
+            static_cast<size_t>(paddings[5]),
+        };
+        break;
+    default:
+        throw std::runtime_error("Unsupported pooling paddings " + std::to_string(paddings.size()));
+    }
+}
+
+NamedOutputs pool3d(const NodeContext& node) {
+    auto data = node.get_input("X");
+
+    auto pooling_type = node.get_attribute<std::string>("pooling_type", {});
+    auto global_pooling = node.get_attribute<bool>("global_pooling");
+    auto adaptive = node.get_attribute<bool>("adaptive");
+    auto kernel_shape = node.get_attribute<std::vector<int32_t>>("ksize");
+
+    auto rounding_type =
+        node.get_attribute<bool>("ceil_mode", false) ? ov::op::RoundingType::CEIL : ov::op::RoundingType::FLOOR;
+
+    if (pooling_type.empty()) {
+        pooling_type = "max";
+    }
+
+    PADDLE_OP_CHECK(node, (pooling_type == "max") || (pooling_type == "avg"), "pool3d: not supported pooling type !");
+    PADDLE_OP_CHECK(node, kernel_shape.size() == 1 || kernel_shape.size() == 3, "pool3d: ksize must be 1 or 3!");
+
+    PartialShape input_shape = data.get_partial_shape();
+
+    int32_t input_rank = static_cast<int32_t>(input_shape.rank().get_length());
+    PADDLE_OP_CHECK(node, input_rank >= 2, "input tensor rank must be greater than 2");
+
+    auto auto_pad = ov::op::PadType::EXPLICIT;
+    ov::Shape pad_begin, pad_end;
+    get_paddings(node, pad_begin, pad_end, auto_pad);
+
+    if (global_pooling || (adaptive && std::any_of(kernel_shape.begin(), kernel_shape.end(), [](int32_t i) {
```

Comment:
Is there test covering case of 'adaptive && one of kernel_shape == 1

Reply:
Already covered